### PR TITLE
Implement new SmallButton type widget

### PR DESCRIPTION
**Pull request by JvnSlv**
#### Overview

This PR implements a new `SmallButton` type, `SmallButtonType.icon`,  for displaying Icon buttons

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [ ] ~Added unit tests for new code and verified existing tests work as expected~

*Code quality*
- [x] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [ ] ~Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations~
- [x] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

File: packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart [hunk 1]
```diff
@@ -122,6 +122,16 @@ class SmallButton extends StatelessWidget {
           onPressed: onPressed,
           child: icon,
         );
+      case SmallButtonType.icon:
+        return IconButton(
+          style: IconButton.styleFrom(
```
Comment (StanevPrime) [inline packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:127]: How the button color can be changed?

File: packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart [hunk 2]
```diff
@@ -122,6 +122,31 @@ class SmallButton extends StatelessWidget {
...
+              width: 0,
+            ),
+          ),
+          color: primaryColor,
+          child: Ink(
+            decoration: ShapeDecoration(
+              color: context.widgetToolkitTheme.smallButtonBackgroundColor,
+              shape: const CircleBorder(),
+            ),
+            child: IconButton(
```
Comment (StanevPrime) [inline packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:139]: What's the purpose of the widgets above ? Do we really need them ?

Comment (JvnSlv) [reply packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:139]: Yes, those are used in order to change the background color of the `IconButton`
`IconButton` has a property `color` but it's used to change the color of the icon.

Example of colored IconButton here
https://api.flutter.dev/flutter/material/IconButton-class.html#material.IconButton.2

Comment (StanevPrime) [reply packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:139]: Please add golden tests with a two colors in light and dark mode

**Review by StanevPrime — CHANGES_REQUESTED** [review]
Please add golden tests with a two colors in light and dark mode

File: packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart (hunk 2 above)
Comment (JvnSlv) [reply packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:139]: This does nothing to the button unfortunately
Changing both background/foreground color doesn't change the background of the button

Comment (StanevPrime) [reply packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:139]: Looking at the official documentation it looks like we have added extra widgets. 

Let's try to control the following colors

1. Background color
2. Icon color

**Official documentation**

```dart
 Material(
      color: Colors.white,
      child: Ink(
          decoration: const ShapeDecoration(
            color: Colors.red, //background color
            shape: CircleBorder(),
          ),
          child: IconButton(
            icon: const Icon(Icons.android),
            color: Colors.white, // icon color
          ),
        ),
    );
```

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart (hunk 2 above)
Comment (JvnSlv) [reply packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart]: It seems like it isn't necessary to use Material and Ink widgets after all, the default styling was being overwritten by by child widget of the IconButton

File: packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart [hunk 3]
```diff
@@ -74,7 +74,7 @@ class SmallButton extends StatelessWidget {
       padding: context.widgetToolkitTheme.smallButtonPadding,
       decoration: BoxDecoration(
         color: state != ButtonStateModel.disabled
-            ? context.widgetToolkitTheme.smallButtonBackgroundColor
+            ? colorStyle?.backgroundColor
```
Comment (StanevPrime) [inline packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:77]: Make sure no breaking changes are introduced. If that's needed the major version of the package needs to be increased

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart (hunk 3 above)
Comment (JvnSlv) [reply packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart:77]: I wouldn't consider this a breaking change, I'm just adding the ability to use the color from `colorStyle`. This should break any existing functionality

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**